src/opensynth/models/faraday/new_gmm/new_gmm_model.py

```diff
@@ -205,11 +205,13 @@ def update_params(
         means: torch.Tensor,
         precision_cholesky: torch.Tensor,
         covariances: torch.Tensor,
+        log_prob: torch.Tensor,
     ):
         self.weights.data = weights
         self.means.data = means
         self.precision_cholesky.data = precision_cholesky
         self.covariances = covariances
+        self.log_prob = log_prob
```

Comment:
You should also register this as buffer too like what we've done for weights, precision_cholesky etc.

Basically states that we want the model to capture.

Reply:
On the same note - can you also store the covariance as a buffer too (missed that in your previous PR).